src/middlewares/autorization/keycloak-auth-handler.ts

```diff
@@ -118,13 +118,60 @@ class KeycloakAuth implements AuthorizationStrategy {
 	}
 
 	async getClient() {
-		const issuer = await Issuer.discover(config.authServerUrl + '/.well-known/openid-configuration');
-		const client = new issuer.Client({
-			client_id: config.clientId,
-			client_secret: config.clientSecret,
-			scope: 'openid',
+		try {
+			const issuer = await Issuer.discover(config.authServerUrl + '/.well-known/openid-configuration');
+			const client = new issuer.Client({
+				client_id: config.clientId,
+				client_secret: config.clientSecret,
+				scope: 'openid',
+			});
+			return client;
+		}catch(e){
+			console.log(e);
+			throw new NetworkError('KEYCLOAK connection failure', 500);
+		}
+	}
+
+	async checkApiKey(apiKey: string){
+		try {
+			const basicAuth = Buffer.from(config.clientId + ':' + config.clientSecret, 'binary').toString('base64');
+			const headers = {
+				headers: {
+					'Content-Type': 'multipart/form-data',
+					Authorization: 'Basic ' + basicAuth,
+				},
+			};
+			const data = new FormData();
+			data.append('apiKey', apiKey);
+			const response = await axios.post(config.authServerUrl + '/apikey/check_api_key/', data, headers);
+			return response;
+		}catch(e){
+			console.log(e);
+			throw new NetworkError(`KEYCLOAK connection failure. Caused by: ${e}`, 500);
+		}
+	}
+
+	 validateIntrospectionResponse(introspectionResponse: any){
+		// Expected structure of keycloak permissions from TokenIntrospectionResponse
+		const KeycloakPermission = z.object({
+			scopes: z.string().array().optional(),
+			rsid: z.string(),
+			rsname: z.string(),
 		});
-		return client;
+
+		type KeycloakPermission = z.infer<typeof KeycloakPermission>;
+
+		const KeycloakIntorspectionAuthorization = z.object({
+			permissions: z.array(KeycloakPermission)
+		})
+		type KeycloakIntorspectionAuthorization = z.infer<typeof KeycloakIntorspectionAuthorization>;
```

Comment:
I think we can move our types out of this function into the top level of this module. No need to have them initialized everytime this function is called.